//
// This file is part of the WoW++ project.
// 
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 2 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program; if not, write to the Free Software 
// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
//
// World of Warcraft, and all World of Warcraft or Warcraft art, images,
// and lore are copyrighted by Blizzard Entertainment, Inc.
// 

syntax = "proto2";
package wowpp.proto;

message LevelStats {
	optional uint32 stat1 = 1;
	optional uint32 stat2 = 2;
	optional uint32 stat3 = 3;
	optional uint32 stat4 = 4;
	optional uint32 stat5 = 5;
	optional float regenhp = 6;
	optional float regenmp = 7;
}

message ClassLevelStats {
	map<uint32, LevelStats> stats = 1;
}

message LevelEntry {
	required uint32 id = 1;
	required uint32 nextlevelxp = 2;
	required uint32 explorationbasexp = 3;
	// [race-id][class-id].stat1 etc.
	map<uint32, ClassLevelStats> stats = 4;
}

message Levels {
	repeated LevelEntry levels = 1;
}
